unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/PlayerInfoCardHUD/PlayerInfoCardHUDController.cs

```diff
@@ -149,17 +150,23 @@ private void SetUserProfile(UserProfile userProfile)
     {
         Assert.IsTrue(userProfile != null, "userProfile can't be null");
 
-        view.SetName(FilterName(userProfile));
-        view.SetDescription(FilterDescription(userProfile));
+        AsyncSetUserProfile(userProfile).Forget();
+    }
+    private async UniTaskVoid AsyncSetUserProfile(UserProfile userProfile)
+    {
+        view.SetName(await FilterName(userProfile));
+        view.SetDescription(await FilterDescription(userProfile));
         view.ClearCollectibles();
         view.SetIsBlocked(IsBlocked(userProfile.userId));
         LoadAndShowWearables(userProfile);
         UpdateFriendshipInteraction();
 
         if (viewingUserProfile != null)
             viewingUserProfile.snapshotObserver.RemoveListener(view.SetFaceSnapshot);
+
         userProfile.snapshotObserver.AddListener(view.SetFaceSnapshot);
         viewingUserProfile = userProfile;
+        view.SetCardActive(true);
```

Comment:
changing the card activation introduces a cohesion problem.. perhaps you could move it outside this method by adding an `Action onFinished` parameter

Reply:
Yeah you are right, I've moved this where it was before but it will wait for the update first